test/e2e/external_oidc_test.go

```diff
@@ -2,54 +2,244 @@ package e2e
+	t.Run("invalid OIDC config degrades auth operator", func(t *testing.T) {
+		for _, tt := range []struct {
+			name       string
+			specUpdate func(*configv1.AuthenticationSpec)
+		}{
+			{"invalid issuer CA bundle", func(s *configv1.AuthenticationSpec) {
+				s.OIDCProviders[0].Issuer.CertificateAuthority.Name = "invalid-ca-bundle"
+			}},
+			{"invalid issuer URL", func(s *configv1.AuthenticationSpec) {
+				s.OIDCProviders[0].Issuer.URL = "https://invalid-idp.testing"
+			}},
+		} {
+			t.Run(tt.name, func(t *testing.T) {
+				err := tc.authResourceRollback(testCtx, origAuthSpec)
+				require.NoError(t, err, "failed to roll back auth resource")
+
+				tc.updateAuthResource(t, testCtx, testSpec, tt.specUpdate)
+
+				require.NoError(t, test.WaitForClusterOperatorDegraded(t, tc.configClient.ConfigV1(), "authentication"))
+			})
+		}
+	})
+
+	t.Run("invalid OIDC config crashes kube-apiserver pods during rollout", func(t *testing.T) {
+		tc.updateAuthResource(t, testCtx, testSpec, func(baseSpec *configv1.AuthenticationSpec) {
+			baseSpec.OIDCProviders[0].ClaimMappings.Username.Claim = ""
+		})
+
+		require.NoError(t, test.WaitForClusterOperatorStatusAlwaysAvailable(t, testCtx, tc.configClient.ConfigV1(), "authentication"))
+		require.NoError(t, test.WaitForClusterOperatorStatusAlwaysProgressing(t, testCtx, tc.configClient.ConfigV1(), "kube-apiserver"))
+
+		tc.requireKASRolloutCrash(t, testCtx, nil, 0)
+	})
+
+	t.Run("OIDC config rolls out successfully but breaks authentication when username claim is unknown", func(t *testing.T) {
+		kasOriginalRevision := tc.kasLatestAvailableRevision(t, testCtx)
+		auth := tc.updateAuthResource(t, testCtx, testSpec, func(baseSpec *configv1.AuthenticationSpec) {
+			baseSpec.OIDCProviders[0].ClaimMappings.Username = configv1.UsernameClaimMapping{
+				TokenClaimMapping: configv1.TokenClaimMapping{
+					Claim: "unknown",
+				},
+				PrefixPolicy: configv1.NoPrefix,
+				Prefix:       nil,
+			}
+		})
+
+		require.NoError(t, test.WaitForClusterOperatorStatusAlwaysAvailable(t, testCtx, tc.configClient.ConfigV1(), "authentication"))
+		require.NoError(t, test.WaitForClusterOperatorStatusAlwaysAvailable(t, testCtx, tc.configClient.ConfigV1(), "kube-apiserver"))
+
+		tc.requireKASRolloutSuccessful(t, testCtx, &auth.Spec, kasOriginalRevision, "")
+
+		tc.testOIDCAuthentication(t, testCtx, kcClient, "", "", false)
+	})
+
+	t.Run("OIDC config rolls out successfully", func(t *testing.T) {
+		for _, tt := range []struct {
+			claim          string
+			prefixPolicy   configv1.UsernamePrefixPolicy
+			prefix         *configv1.UsernamePrefix
+			expectedPrefix string
+		}{
+			{"email", configv1.Prefix, &configv1.UsernamePrefix{PrefixString: "oidc-test:"}, "oidc-test:"},
+			{"email", configv1.NoPrefix, nil, ""},
+			{"sub", configv1.NoOpinion, nil, idpURL + "#"},
+			{"email", configv1.NoOpinion, nil, ""},
+		} {
+			policyStr := "NoOpinion"
+			if len(tt.prefixPolicy) > 0 {
+				policyStr = string(tt.prefixPolicy)
+			}
```

Comment:
Just set it to the prefix policy defined in the test case? All existing test cases specify the prefix policy explicitly and I can't think of a scenario where it would make sense to define a test without an explicit policy.

Reply:
Well, `configv1.NoOpinion` is an empty string; I added this explicitly because the test name would read weirdly as "username_claim_email_prefix_policy_".

Reply:
Ahhhhh - Gotcha. Thanks for the clarification there. I missed that :). Feel free to ignore